src/PowerShellEditorServices/Utility/ExecutionTimer.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace Microsoft.PowerShell.EditorServices.Utility
+{
+    /// <summary>
+    /// Simple timer to be used with `using` to time executions.
+    /// </summary>
+    /// <example>
+    /// An example showing how ExecutionTimer is intended to be used
+    /// <code>
+    /// using (ExecutionTimer.Start(logger, "Execution of MyMethod completed."))
+    /// {
+    ///     MyMethod(various, arguments);
+    /// }
+    /// </code>
+    /// This will print a message like "Execution of MyMethod completed. [50ms]" to the logs.
+    /// </example>
+    public struct ExecutionTimer : IDisposable
+    {
+        [ThreadStatic]
+        private static Stopwatch s_stopwatch;
```

Comment:
Why not an instance member?  And are we sure this will always be created and disposed of on the same thread?  After an `await` expression, the thread for the rest of the method could be different.

Reply:
Or I guess it wouldn't matter since the expression generated by the compiler would save it to a temporary variable...

Yeah never mind this one, marking it resolved.